assembler/parse.c

```diff
@@ -81,6 +81,38 @@ int check_legal_comma(char* str, int index)
 	return COMMA_OK;
 }
 
+
+void split_line(char* str, int index)
+{
+    int splitPlace = 0;
+    
+    while (!is_end_of_line(str[index]))
+    {
+        if (is_space(str[index]))
+		{
+	    index = clear_white_space(str, index);
+        if (is_end_of_line(str[index]))
+		break;
+		}
+        if (str[index] == ',')
+        {
+           index++;
+        }
+            
+        splitLine[splitPlace] = &str[index]; /*points to the start of a word*/
+
+        while ( (!is_space(str[index])) && (str[index] != ',') && (!is_end_of_line(str[index]))) /*skip to the end of the word */
```

Comment:
Can the function "clear_word " in parse.h file help you instead of the long while?